[Lee Zhi Xuan] iP

itszhixuan opened the pull request.

itszhixuan committed:
Level 0. Greet

itszhixuan committed:
Level 1. Greet, Echo, Exit

itszhixuan committed:
Level 2. Add, List

itszhixuan committed:
Level 3. Mark as Done

itszhixuan committed:
Level-3 A-Classes

itszhixuan committed:
A-CodingStandard

itszhixuan committed:
Level 4. ToDos, Events, Deadlines

sansders commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,104 @@
+
+    private static void processInputs() {
+        Scanner in = new Scanner (System.in);
+        Task[] storeValues = new Task[MAX_TASK_NUM];
```
Would probably be better to name this as `storedValues` instead

sansders commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,104 @@
+        formattingLine();
+        System.out.println("added: " + line);
+        formattingLine();
+        //store value in line into list
```
Need to add whitespace at the start of the comment, and begin with comment with `Stores` instead of `store`

sansders commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,104 @@
+        int length = line.length();
+        String itemToMark = line.substring(7, length);
+        int numToMark = Integer.parseInt(itemToMark);
+        //unmark the item
```
Need to change every other comment to adhere to the proper coding standard

sansders commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,104 @@
+import java.util.Arrays;
 public class Duke {
+    public static final int MAX_TASK_NUM = 100;
     public static void main(String[] args) {
```
The main method looks good and clean 👍

sansders reviewed (commented):
Overall, very clean code!

heejet commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,104 @@
     public static void main(String[] args) {
+        showGreetings();
+        processInputs();
+        showGoodbye();
```
Good use of SLAP here to make the code readable.

heejet commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,104 @@
+                unmarkItem(storeValues, line);
+            } else {
+                counter = addTask(storeValues, counter, line);
+            }
```
Perhaps you can consider using a switch statement here instead.

heejet commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,104 @@
+    private static void processInputs() {
+        Scanner in = new Scanner (System.in);
+        Task[] storeValues = new Task[MAX_TASK_NUM];
+        int counter = 0;
```
Maybe you can consider renaming this variable to numTasks or something more descriptive to enhance readability.

heejet commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,104 @@
+
+    private static void unmarkItem(Task[] storeValues, String line) {
+        int length = line.length();
+        String itemToMark = line.substring(7, length);
```
Consider switching the magic number 7 to a constant to make the code easier to understand.

heejet commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,104 @@
+    private static void formattingLine() {
+        String lineFormatting = "____________________________________________________________\n";
+        System.out.println(lineFormatting);
     }
```
I like how you extract out these codes into methods to make it more readable and reduce code duplication.

heejet commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,104 @@
+    private static void showGoodbye() {
+        String bye = "____________________________________________________________\n" +
+                "Bye. Hope to see you again soon!\n" +
+                "____________________________________________________________\n";
```
Perhaps you could use your formattingLine() method here to make your code more readable.

heejet commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,104 @@
+    }
+
+    private static void formattingLine() {
+        String lineFormatting = "____________________________________________________________\n";
```
Consider setting this as a constant variable so that the length of the line is standardized across all methods.

itszhixuan committed:
Added comments and removed magic numbers

itszhixuan committed:
A-TextUiTesting

itszhixuan committed:
initial attempt to catch errors

itszhixuan committed:
convert if-else loop to switch-case as suggested in peer evaluation

itszhixuan committed:
Level-5: Handle Errors

itszhixuan committed:
Merge branch 'branch-Level-5'

itszhixuan committed:
Update exceptions

itszhixuan committed:
Using ArrayList<Task> to store values instead

itszhixuan committed:
Added delete feature into Duke

itszhixuan committed:
Error catching for delete

itszhixuan committed:
Merge branch 'branch-Level-6'

itszhixuan committed:
Todo, Deadline and Event appends initial input to txt file

itszhixuan committed:
Duke processes inputs that were recorded in txt file in the previous interaction.

Currently still missing: updating mark in txt file when called and deleting item from txt file (after merging with branch 6)

itszhixuan committed:
mark and unmark updates the txt.file

itszhixuan committed:
code will create file on its own if it does not exist

itszhixuan committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Duke.java

itszhixuan committed:
Delete function works with txt file. Level-7

itszhixuan committed:
Refactor variables

cheehengk commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,50 @@
+
+    /***
+     * Gets the current status of the task within the list.
+     * @return isDone.
```
Good effort in starting on commenting your code. Perhaps be more specific in what this method returns, rather than just the name of the variable being returned.

cheehengk commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,603 @@
+import java.io.BufferedReader;
+import java.io.IOException;
+
 public class Duke {
```
Duke class is very long, try extracting to have smaller classes to handle different things.

cheehengk commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,603 @@
+            String task = extractType[4];
+
+            switch (type) {
+            case "T":
+                readExistingToDo(storedValues, marked, task);
+                break;
+            case "D":
+                readExistingDeadline(storedValues, extractType, marked);
+                break;
+            case "E":
+                readExistingEvent(storedValues, extractType, marked);
+                break;
```
Remember to have default path in switch-cases.

cheehengk commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,603 @@
+        switch (command) {
+        case "bye":
+            return true;
+        case "list":
```
Could we use constants here instead of raw string here?

cheehengk commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,603 @@
+     * @param extractType Description of task with duration.
+     * @param marked Status of marked in txt file.
+     */
+    private static void readExistingEvent(ArrayList<Task> storedValues, String[] extractType, String marked) {
+        int firstSlash = extractType[4].indexOf("|");
+        String eventName = extractType[4].substring(0,firstSlash-1);
+        String eventDuration = extractType[4].substring(firstSlash+2);
+        int secondSlash = eventDuration.indexOf("|");
+        String eventFrom = eventDuration.substring(0,secondSlash-1);
+        String eventTo = eventDuration.substring(secondSlash+2);
+        Event addInputE = new Event(eventName, eventFrom, eventTo);
+        storedValues.add(taskNum,addInputE);
+        if (marked.equals("1")) {
+            storedValues.get(taskNum).markAsDone();
+        }
```
Some repeated code here with readExistingDeadline() method. Perhaps think if it is possible to write smaller methods and reuse them?

cheehengk commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,603 @@
+     * @param numToMark The row in txt file to be unmarked.
+     * @throws IOException Thrown when file cannot be read.
+     */
+    private static void unmarkTaskInTxt(ArrayList<Task> storedValues, int numToMark) throws IOException {
```
This is a long method doing many things. Try having separate method handling different tasks. Helps with debugging too :)

cheehengk reviewed (commented):
Overall a good effort in implementation. Some long methods and classes here and there. Some work on SLAP would go a long way.

itszhixuan committed:
Refactoring writing and deleting functions into Storage.java file

itszhixuan committed:
Refactor functions that edit the task list into TaskList.java

itszhixuan committed:
Categorising mark commands to txt into Storage.java file as well

itszhixuan committed:
Moved commands that makes sense of user inputs into Parser,java

itszhixuan committed:
Moved commands relating to user input into Ui, and updated code in Storage.java

itszhixuan committed:
Finalise refactoring OOP

itszhixuan committed:
Implemented find feature where the list will appear given the keyword

itszhixuan committed:
Added the remaining JavaDoc comments for the new functions added

itszhixuan committed:
Edited according to comments given by Chee Heng, excluding last comment on unmarkTaskInTxt

itszhixuan committed:
Simplified unmarkTaskInTxt file

itszhixuan committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/Parser.java
#	src/main/java/Ui.java

itszhixuan committed:
All JavaDoc comments are completed for new variables added.

itszhixuan committed:
Merge pull request #2 from itszhixuan/branch-A-JavaDoc

All JavaDoc comments are completed for new variables added.

itszhixuan committed:
Created README document that serves as the user guide

itszhixuan committed:
Missing fixed constant for "find"

itszhixuan committed:
Change file path and remove java.io.*

itszhixuan committed:
Catch index out of bounds error for mark and unmark
